### PR TITLE
Feature/geode 11 gfsh commands

### PR DESCRIPTION
**Pull request by aparnard**
GEODE-11: Implementing GFSH command to list Lucene indexes

Added a GFSH command to list the Lucene indexes.  Also added a dunit test to verify this functionality. The org.springframework.shell.core.CommandMarker file is required to register the Lucene commands in GFSH. The GFSH command to list the Lucene indexes is implemented through the LuceneIndexCommands class and the LuceneListIndexFunction class gets information about the Lucene indexes from the cache.

### CODE REVIEW

**Comment by upthewaterspout** [comment]
+1

**Comment by jhuynh1** [comment]
+1

nit picky- there is a comment with the word lucene that should probably be capitalized.  Other than that looks good

File: geode-lucene/src/main/java/com/gemstone/gemfire/cache/lucene/internal/cli/LuceneCliStrings.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.gemstone.gemfire.cache.lucene.internal.cli;
+
+public class LuceneCliStrings {
+  public static final String LUCENE_LIST_INDEX = "lucene list indexes";
+  public static final String LUCENE_LIST_INDEX__HELP = "Display the list of lucene indexes created for all members.";
```
Comment (nabarunnag) [inline geode-lucene/src/main/java/com/gemstone/gemfire/cache/lucene/internal/cli/LuceneCliStrings.java]: Capitalization of Lucene and Index is bit inconsistent in these strings.

File: geode-lucene/src/test/java/com/gemstone/gemfire/cache/lucene/internal/cli/LuceneIndexCommandsDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import org.junit.experimental.categories.Category;
+
+import static com.gemstone.gemfire.cache.lucene.test.LuceneTestUtilities.*;
+import static com.gemstone.gemfire.test.dunit.Assert.*;
+
+@Category(DistributedTest.class)
+public class LuceneIndexCommandsDUnitTest extends CliCommandTestBase {
+
+  @Test
+  public void testListIndex() throws Exception {
```
Comment (nabarunnag) [inline geode-lucene/src/test/java/com/gemstone/gemfire/cache/lucene/internal/cli/LuceneIndexCommandsDUnitTest.java]: I am not sure of this one but should we maintain the test naming convention as test<what is expected><what is test situation or what is passed>()

**Comment by nabarunnag** [comment]
+1

**Comment by jinmeiliao** [comment]
+1

**Comment by gesterzhou** [comment]
+1

I will push it to develop

**Pull request closed without merging**